### PR TITLE
Will use logicalDestPort only in FlowClassifier

### PR DESCRIPTION
**Pull request by dmitryintel**
* Integration tests pass with dummy sfc driver (not ovs)
* The driver is set in /etc/neutron/neutron.conf under [sfc] section.
* Use the line 'drivers = dummy' instead of 'drivers = ovs'.

### CODE REVIEW

File: nsfc-plugin/src/test/java/org/osc/controller/nsfc/NeutronSfcSdnRedirectionApiTest.java [hunk 1]
```diff
@@ -69,6 +72,18 @@ public void setup() throws Exception {
...
...xt_request_id\": \"req-3cb168df-cb16-4a62-9a38-5d1f64f6aabf\", \"timestamp\": \"2018-03-01 18:10:48.031575\", \"_context_tenant_id\": \"deddd82fd0d746a8b9dbbee3f4bdb9f0\", \"_context_user\": \"57c280566a694377acd53e944acba7f2\", \"_unique_id\": \"0e48ddac9c874cf5a966f62ed92a430c\", \"_context_resource_uuid\": null, \"_context_tenant_name\": \"demo\", \"_context_user_id\": \"57c280566a694377acd53e944acba7f2\", \"payload\": {\"port_id\": \"c81ff7f3-a5cf-4fb7-a9d5-234623357045\"}, \"_context_project_name\": \"demo\", \"_context_user_identity\": \"57c280566a694377acd53e944acba7f2 deddd82fd0d746a8b9dbbee3f4bdb9f0 - - -\", \"_context_auth_token\": \"ceedb2ca2e7a40609abc9b4a3f797542\", \"_context_show_deleted\": false, \"_context_tenant\": \"deddd82fd0d746a8b9dbbee3f4bdb9f0\", \"priority\": \"INFO\", \"_context_read_only\": false, \"_context_is_admin\": true, \"_context_project_id\": \"deddd82fd0d746a8b9dbbee3f4bdb9f0\", \"_context_project_domain\": null, \"_context_timestamp\": \"2018-03-01 18:10:47.833611\", \"_context_user_domain\": null, \"_context_user_name\": \"admin\", \"publisher_id\": \"network.controller1.localdomain\", \"message_id\": \"90fe9a6c-ccd7-4707-a153-f31a8a0854ec\"}";
```
Comment (arvindn05) [inline nsfc-plugin/src/test/java/org/osc/controller/nsfc/NeutronSfcSdnRedirectionApiTest.java]: can we put into a resource file?

also not used?

File: nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java [hunk 2]
```diff
@@ -265,7 +267,10 @@ public void setup() {
 
         this.osClient = v3.authenticate();
 
-        LOG.debug("You should have prepared an opentack with sfc and two servers with two ports each!");
+        LOG.debug("You should have prepared an opentack with sfc and two servers on management network!");
```
Comment (arvindn05) [inline nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java]: i would remove these messages. we should not document with log messages

File: nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java [hunk 3]
```diff
@@ -265,7 +267,10 @@ public void setup() {
 
         this.osClient = v3.authenticate();
```
Comment (arvindn05) [inline nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java]: i am confused. does the osgi integration test require a working openstack installation? This is not the case for other OSGI integration tests. i would setup a dummy server via java code.

see https://github.com/arvindn05/security-mgr-nsm-plugin/blob/master/src/test/java/com/mcafee/nsm/AbstractNsmTest.java

**Review by arvindn05 — CHANGES_REQUESTED** [review]

File: nsfc-plugin/src/test/java/org/osc/controller/nsfc/LiveEnvIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+import org.osc.sdk.controller.api.SdnControllerApi;
+import org.osc.sdk.controller.api.SdnRedirectionApi;
+import org.osc.sdk.controller.element.Element;
+import org.osc.sdk.controller.element.InspectionHookElement;
+import org.osc.sdk.controller.element.NetworkElement;
+import org.osc.sdk.controller.element.VirtualizationConnectorElement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LiveEnvIntegrationTest {
```
Comment (arvindn05) [inline nsfc-plugin/src/test/java/org/osc/controller/nsfc/LiveEnvIntegrationTest.java]: comments or documentation on what this does and what needs to be available/attr updated for a successful run

File: nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java [hunk 5]
```diff
@@ -251,234 +115,8 @@ public SSLContext getSslContext() {
...
-        assertEquals(inspectionPortElement.getEgressPort().getElementId(), portPairCheck.getEgressId());
-    }
-
-    private void assertPortPairGroupIsOk(PortPairElement inspectionPortElement) {
-        assertNotNull(inspectionPortElement.getPortPairGroup());
-        PortPairGroup ppgCheck = this.osClient.sfc().portpairgroups().get(inspectionPortElement.getParentId());
-        assertNotNull(ppgCheck);
-        assertNotNull(ppgCheck.getPortPairs());
-        assertTrue(ppgCheck.getPortPairs().contains(inspectionPortElement.getElementId()));
+       assertNotNull(this.api);
```
Comment (arvindn05) [inline nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java]: does this fail when unresolved osgi libraries exist?

File: nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java [hunk 6]
```diff
@@ -251,234 +115,8 @@ public SSLContext getSslContext() {
...
-    public void tearDown() throws Exception {
-        if (this.redirApi != null) {
-            this.redirApi.close();
-        }
-        cleanAllOnOpenstack();
-    }
-
-//  @Test
+    @Test
     public void verifyApiResponds() throws Exception {
```
Comment (arvindn05) [inline nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java]: also update name

**Review by arvindn05 — CHANGES_REQUESTED** [review]

File: nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java (hunk 5 above)
Comment (dmitryintel) [reply nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java]: Some potential class-not-found exceptions are not detected, but only runtime ones.

File: nsfc-plugin/src/test/java/org/osc/controller/nsfc/LiveEnvIntegrationTest.java (hunk 4 above)
Comment (dmitryintel) [reply nsfc-plugin/src/test/java/org/osc/controller/nsfc/LiveEnvIntegrationTest.java]: Added a javadoc

**Review by arvindn05 — APPROVED** [review]

**Pull request merged**